static/app/views/insights/common/queries/useDiscover.ts

```diff
@@ -97,6 +99,18 @@ export const useMetrics = <Fields extends MetricsProperty[]>(
   );
 };
 
+export const useDiscoverDataset = <Fields extends DiscoverProperty[]>(
+  options: UseDiscoverOptions<Fields> = {},
+  referrer: string
+) => {
+  const useEap = useInsightsEap();
+  return useDiscover<Fields, DiscoverResponse>(
+    options,
+    useEap ? DiscoverDatasets.SPANS_EAP_RPC : DiscoverDatasets.DISCOVER,
+    referrer
+  );
+};
+
```

Comment:
I'm a little confused by this hook, why does it make switching easier? The name `useDiscoverDataset` implies that it returns a dataset, but it doesn't. The types say that the fields must be a `DiscoverProperty`, but it might return either a `DISCOVER` response or a `SPANS_EAP_RPC` response, which is confusing. Plus, one of the places it's used, it only fetches indexes spans anyway. Is this really better than continuing to use `useEap` next to `useDiscover` everywhere?

Reply:
I don't the name is that important as it's getting remove right after we migrate. There's already a `useDiscover` hook and I didn't want to change that only to change it back once we migrate.

But I do agree that it's better to just call`useEapSpans` instead